### ISSUE TITLE
What is the function in `self.model_param` when developing  a new algorithm component?

### ISSUE THREAD

**Issue by gxcuit** [FederatedAI/FATE#3498]
**Describe the bug**
Hi, 我在[根据教程](https://fate.readthedocs.io/en/latest/develop/develop_guide)自定义新的component的时候，遇到了一个问题。如果没有在`__init__` 方法中给`self.model_param`赋值的话，则无法通过`params` 获得参数。教程中并没有给出这个步骤，是代码的问题还是教程文档的问题。

下面以我想重新开发`homo_lr` 为例。
**To Reproduce**
1. 复制一份`federatedml/param/logistic_regression_param.py`， 并将其中的`class HomoLogisticParam` 重命名为`class PoisonParam`。其余都不变。
2. 修改conf文件
3. 如果仅做这些修改，打印的log显示，params的类型没有改变，并且参数值未传递
`[DEBUG] [2021-12-02 10:12:42,083] [5866:140442401756992] - homo_lr_host_mp.py[line:65]: params is <federatedml.param.logistic_regression_param.HomoLogisticParam object at 0x7fbb34c8b358>
`
但component已经收到数据，并且类型为PoisonParam
`[DEBUG] [2021-12-02 10:12:41,319] [5866:140442401756992] - component_properties.py[line:73]: {'PoisonParam': {'penalty': 'L2', 'tol': 1e-05, 'alpha': 0.01, 'optimizer': 'sgd', 'batch_size': -1, 'learning_rate': 0.15, 'init_param': {'init_method': 'zeros', 'init_const': 1, 'fit_intercept': True, 'random_seed': None}, 'max_iter': 18, 'early_stop': 'diff', 'encrypt_param': {'method': None, 'key_length': 1024}, 'predict_param': {'threshold': 0.5}, 'cv_param': {'n_splits': 5, 'mode': 'hetero', 'role': 'guest', 'shuffle': True, 'random_seed': 1, 'need_cv': False, 'output_fold_history': True, 'history_value_type': 'score'}, 'decay': 1, 'decay_sqrt': True, 'multi_class': 'ovr', 'validation_freqs': 1, 'stepwise_param': {'score_name': 'AIC', 'mode': 'hetero', 'role': 'guest', 'direction': 'both', 'max_step': 10, 'nvmin': 2, 'nvmax': None, 'need_stepwise': False}, 'early_stopping_rounds': None, 'metrics': ['auc', 'ks'], 'use_first_metric_only': False, 'floating_point_precision': 23, 're_encrypt_batches': 2, 'aggregate_iters': 1, 'use_proximal': False, 'mu': 0.1, 'poison': True}, 'dsl_version': 2, 'initiator': {'role': 'guest', 'party_id': 10000}, 'role': {'guest': [10000], 'host': [9999], 'arbiter': [9999]}, 'job_parameters': {'job_type': 'train', 'work_mode': 0, 'backend': 0, 'computing_engine': 'STANDALONE', 'federation_engine': 'STANDALONE', 'storage_engine': 'STANDALONE', 'engines_address': {'computing': {'nodes': 1, 'cores_per_node': 20}, 'federation': {'nodes': 1, 'cores_per_node': 20}, 'storage': {'nodes': 1, 'cores_per_node': 20}}, 'federated_mode': 'SINGLE', 'task_parallelism': 1, 'computing_partitions': 4, 'federated_status_collect_type': 'PULL', 'model_id': 'arbiter-9999#guest-10000#host-9999#model', 'model_version': '202112021012005851198', 'eggroll_run': {'eggroll.session.processors.per.node': 4}, 'spark_run': {}, 'rabbitmq_run': {}, 'pulsar_run': {}, 'adaptation_parameters': {'task_nodes': 1, 'task_cores_per_node': 4, 'task_memory_per_node': 0, 'request_task_cores': 4, 'if_initiator_baseline': False}}, 'component_parameters': {'common': {'dataio_0': {'with_label': True, 'output_format': 'dense'}, 'dataio_1': {'with_label': True, 'output_format': 'dense'}, 'homo_lr_0': {'penalty': 'L2', 'tol': 1e-05, 'alpha': 0.01, 'optimizer': 'sgd', 'batch_size': -1, 'learning_rate': 0.15, 'init_param': {'init_method': 'zeros'}, 'max_iter': 18, 'aggregate_iters': 1, 'early_stop': 'diff', 'encrypt_param': {'method': None}, 'poison': True, 'validation_freqs': 1}, 'evaluation_0': {'eval_type': 'binary'}}, 'role': {'guest': {'0': {'reader_0': {'table': {'name': 'breast_homo_guest', 'namespace': 'experiment'}}, 'reader_1': {'table': {'name': 'homo_breast_test', 'namespace': 'experiment'}}}}, 'host': {'0': {'reader_0': {'table': {'name': 'breast_homo_host', 'namespace': 'experiment'}}, 'reader_1': {'table': {'name': 'homo_breast_test', 'namespace': 'experiment'}}, 'evaluation_0': {'need_run': True}}}}}, 'conf_path': 'homo_lr_mp_conf.json', 'dsl_path': 'homo_lr_mp_dsl.json', 'local': {'role': 'host', 'party_id': 9999}, 'CodePath': 'federatedml/linear_model/logistic_regression/homo_logsitic_regression/homo_lr_host_mp.py/HomoLRHost', 'module': 'HomoLRMP', 'output_data_name': ['data']}`

注：尽管component 收到runtime conf的参数，但实际训练的时候仍然使用默认值。

如果将`homo_lr_base.py` 的`__init__`函数中的`self.model_param = HomoLogisticParam()`进行修改，则可解决这个问题。

**Expected behavior**
是否应该在文档中加上修改self.model_param的这个步骤？
**Additional context**

**Comment by mgqa34** [comment]
Yes, you're right,  thanks for pointing out. Can you make a pull request to fix this, or if you have no time, we will fix it later.

**Comment by gxcuit** [comment]
> Yes, you're right, thanks for pointing out. Can you make a pull request to fix this, or if you have no time, we will fix it later.

Hi, Thanks for your reply.
I will make a pull request.
